backend/app/core/database.py

```
@@ -62,16 +62,73 @@ def receive_after_cursor_execute(
-        db.rollback()
+        logger.error(f"Database session error: {e}", exc_info=True)
+        try:
+            # Ensure we rollback any failed transaction
+            db.rollback()
+        except Exception as rollback_error:
+            logger.error(f"Failed to rollback transaction: {rollback_error}")
+            # If rollback fails, close and recreate session
+            try:
+                db.close()
+            except:
+                pass
```

Ruff (0.12.2)

Use `contextlib.suppress(BaseException)` instead of `try`-`except`-`pass`

Replace with `contextlib.suppress(BaseException)`

(SIM105)